tests/integration.rs

```diff
@@ -5712,6 +5738,87 @@ mod tests {
+        #[test]
+        fn test_windows_guest_cpu_hotplug() {
+            let mut windows = WindowsDiskConfig::new(WINDOWS_IMAGE_NAME.to_string());
+            let guest = Guest::new(&mut windows);
+
+            let tmp_dir = TempDir::new_with_prefix("/tmp/ch").unwrap();
+            let mut workload_path = dirs::home_dir().unwrap();
+            workload_path.push("workloads");
+
+            let mut ovmf_path = workload_path.clone();
+            ovmf_path.push(OVMF_NAME);
+
+            let mut osdisk_path = workload_path;
+            osdisk_path.push(WINDOWS_IMAGE_NAME.to_string());
+
+            let api_socket = temp_api_path(&tmp_dir);
+
+            let mut child = GuestCommand::new(&guest)
+                .args(&["--api-socket", &api_socket])
+                .args(&["--cpus", "boot=2,max=8,kvm_hyperv=on"])
+                .args(&["--memory", "size=4G"])
+                .args(&["--kernel", ovmf_path.to_str().unwrap()])
+                .default_disks()
+                .args(&["--serial", "tty"])
+                .args(&["--console", "off"])
+                .args(&["--net", "tap="])
+                .capture_output()
+                .spawn()
+                .unwrap();
+
+            // Wait to make sure Windows boots up
+            thread::sleep(std::time::Duration::new(60, 0));
+
+            let r = std::panic::catch_unwind(|| {
+                let vcpu_num = 2;
+                // Check the initial number of CPUs the guest sees
+                assert_eq!(get_cpu_count_windows(), vcpu_num);
+                // Check the initial number of vcpu threads in the CH process
+                assert_eq!(get_vcpu_threads_count(child.id()), vcpu_num);
+
+                let vcpu_num = 6;
+                // Hotplug some CPUs
+                resize_command(&api_socket, Some(vcpu_num), None, None);
+                // Wait to make sure CPUs are added
+                thread::sleep(std::time::Duration::new(10, 0));
+                // Check the guest sees the correct number
+                assert_eq!(get_cpu_count_windows(), vcpu_num);
+                // Check the CH process has the correct number of vcpu threads
+                assert_eq!(get_vcpu_threads_count(child.id()), vcpu_num);
+
+                /* TODO Uncomment this block once hot-remove is worky.
+                 * Right now this is supposed to fail.
+                let vcpu_num = 4;
+                // Unplug some CPUs
+                resize_command(&api_socket, Some(vcpu_num), None, None);
+                // Wait to make sure CPUs are unplugged
+                thread::sleep(std::time::Duration::new(10, 0));
+                // Check the CH process has the correct number of vcpu threads
+                // Check the guest sees the correct number
+                assert_eq!(get_cpu_count_windows(), vcpu_num);
+                // Check the CH process has the correct number of vcpu threads
+                assert_eq!(get_vcpu_threads_count(child.id()), vcpu_num);*/
```

Comment:
Do you know why it's not working? Have you tried with latest master branch? We merged a fix today about CPU hotunplug.

Reply:
I filed #2457. I'll rebuild right now and rerun the test, lets see if i was to rash.

Thanks.

Reply:
Unfortunately no improvement after the rebase. Seems it'll need another round in the mentioned issue.

Thanks.

Reply:
Thanks for trying :)

Reply:
I don't like having commented out code.

Reply:
ACK, removed that one.

Thanks.